tests/login/ready.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "pushd data"
+    rlPhaseEnd
+
+    for method in ${METHODS:-virtual}; do
+        rlPhaseStartTest "Positive login test for ($method)"
+            rlRun -s "tmt run -a provision -h $method login -c exit" 0-255
+            rlAssertGrep "login: Starting interactive shell" "$rlRun_LOG"
+        rlPhaseEnd
+
+        if [[ $method == "virtual" ]]; then
+            image_url="FOOOOO"
+            rlPhaseStartTest "Negative login test for $method (image url = $image_url)"
+                rlRun -s "tmt run -a provision -h $method -i $image_url login -c exit" 0-255 \
+                    "disallowed to login into guest which is virtual if image url is invalid"
+                rlAssertNotGrep "login: Starting interactive shell" "$rlRun_LOG"
+                rlAssertGrep "Could not map.*to compose" "$rlRun_LOG"
+            rlPhaseEnd
+
+            image_url="file:///rubbish"
```

Comment:
What is the point of the second test? Both `FOOOOO` and `file:///rubbish` are invalid images. Why doing the check twice?

Reply:
It is because `FOOOOO` and `file:///rubbish` throw different errors, e.g.

* 01 - **FOOOOO**  - `Could not map 'fooooo' to compose.`
```bash
$ tmt run -a provision -h virtual -i FOOOOO login
#...<snip>............................................................
    provision
        how: virtual
        user: root
        image: FOOOOO
        memory: 2048 MB
        disk: 10 GB
        arch: x86_64
    fail: Could not map 'fooooo' to compose.
    finish
        summary: 1 task completed
Could not map 'fooooo' to compose.
```

* 02 - **file:///rubbish** - `Image '/var/tmp/tmt/testcloud/images/rubbish' not found.`
```bash
$ tmt run -a provision -h virtual -i file:///rubbish login
#...<snip>............................................................
    provision
        how: virtual
        user: root
        image: file:///rubbish
        memory: 2048 MB
        disk: 10 GB
        arch: x86_64
        progress: downloading...
    fail: Image '/var/tmp/tmt/testcloud/images/rubbish' not found.
    finish
        summary: 1 task completed
Image '/var/tmp/tmt/testcloud/images/rubbish' not found.
The exception was caused by the previous exception:
Specified image path /rubbish does not exist.
```

Reply:
Ok, I see. Thanks for clarification.